### ISSUE TITLE
how can I pass logits before I fit and predict model?

### ISSUE THREAD

**Issue by OnlyBelter** [tensorflow/tensorflow#38166]
Since I need to train a model with multiple labels, I need to use loss function `tf.nn.sigmoid_cross_entropy_with_logits`. This function has two parameters: `logits` and `loss`.
Does parameter `logits`is the value of predicted y? How can I pass this value before I compile model? I cannot predict y before I compile and fit model, right?

This is my code:

```
m_part1 = keras.Sequential([keras.layers.Dense(50, activation='tanh', input_shape=[100]), 
                            keras.layers.Dense(30, activation='relu'),
                            keras.layers.Dense(50, activation='tanh'),
                            keras.layers.Dense(100, activation='relu')])
m_part2 = keras.Sequential([keras.layers.Dense(8, input_shape=[100])])
model = keras.Sequential([m_part1, m_part2])

model.compile(optimizer='rmsprop', 
              loss=tf.nn.sigmoid_cross_entropy_with_logits(logits=y_pred), labels=y),   # <---How to figure out y_pred?
              metrics=['accuracy'])
model.fit(x, y, epochs=10, batch_size=32)
y_pred = model.predict(x)  # <--- Now I got y_pred
```

How to calculate `y_pred` even before compiling model?

**Comment by gadagashwini** [comment]
@OnlyBelter, Can you share the full standalone code and also the Tensorflow version which you are using. Thanks

**Comment by OnlyBelter** [comment]
```
import tensorflow as tf
from tensorflow import keras

model = keras.Sequential([keras.layers.Dense(50, activation='tanh', input_shape=[100]), 
                            keras.layers.Dense(30, activation='relu'),
                            keras.layers.Dense(50, activation='tanh'),
                            keras.layers.Dense(100, activation='relu'),
                            keras.layers.Dense(8)])

model.compile(optimizer='rmsprop', 
              loss=tf.nn.sigmoid_cross_entropy_with_logits(logits=y_pred), labels=y),   # <---How to figure out y_pred?
              metrics=['accuracy'])
model.fit(x, y, epochs=10, batch_size=32)
y_pred = model.predict(x)  # <--- Now I got y_pred
```
It's a simple code fragment, I just don't know how to pass `logits`. I searched some examples, all of them need to construct a custom class to wrap loss function. such as this one: https://tensorflow.google.cn/tutorials/generative/cvae

**Comment by gowthamkpr** [comment]
@OnlyBelter Please take a look at this [article](https://towardsdatascience.com/sigmoid-activation-and-binary-crossentropy-a-less-than-perfect-match-b801e130e31) which clearly explains how to pass logits.
Also please post questions related to support on stack overflow as there is a wider community to respond. Thanks!

**Comment by OnlyBelter** [comment]
I got it, thanks a lot! This link can help:
https://stackoverflow.com/questions/61012727/how-can-i-pass-logits-to-sigmoid-cross-entropy-with-logits-before-i-fit-and-pred
